src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -396,17 +405,19 @@ class ErgoWalletActor(settings: ErgoSettings, boxSelector: BoxSelector)
                   AssetUtils.mergeAssets(boxTokens, acc)
                 }
 
-              val inputBoxes = if (inputs.nonEmpty) {
-                inputs
+              val (inputBoxes, filter) = if (inputs.nonEmpty) {
+                //inputs are provided externally, no need for filtering
+                (inputs
```

Comment:
Long methods chaining inside tuple constructor doesn't looks like a readable code -- try move `inputs.map { .. }.toIterator` to a separate variable.

Reply:
done